apps/processing/src/config/env.ts

```diff
@@ -23,13 +23,38 @@ const validationSchema = z.object({
     DATABASE_SCHEMA: z.string().default("public"),
     INDEXER_GRAPHQL_URL: z.string().url(),
     INDEXER_ADMIN_SECRET: z.string(),
-    COINGECKO_API_KEY: z.string(),
-    COINGECKO_API_TYPE: z.enum(["demo", "pro"]).default("pro"),
+    PRICING_SOURCE: z.enum(["dummy", "coingecko"]).default("coingecko"),
     IPFS_GATEWAYS_URL: stringToJSONSchema
         .pipe(z.array(z.string().url()))
         .default('["https://ipfs.io"]'),
 });
 
+const dummyPricingSchema = baseSchema.extend({
+    PRICING_SOURCE: z.literal("dummy"),
+    DUMMY_PRICE: z.coerce.number().optional().default(1),
+});
+
+const coingeckoPricingSchema = baseSchema.extend({
+    PRICING_SOURCE: z.literal("coingecko"),
+    COINGECKO_API_KEY: z.string().min(1),
+    COINGECKO_API_TYPE: z.enum(["demo", "pro"]).default("pro"),
+});
+
+const validationSchema = z
+    .discriminatedUnion("PRICING_SOURCE", [dummyPricingSchema, coingeckoPricingSchema])
+    .transform((val) => {
+        if (val.PRICING_SOURCE === "dummy") {
+            return { pricingSource: val.PRICING_SOURCE, dummyPrice: val.DUMMY_PRICE, ...val };
+        }
+
+        return {
+            pricingSource: val.PRICING_SOURCE,
+            apiKey: val.COINGECKO_API_KEY,
+            apiType: val.COINGECKO_API_TYPE,
+            ...val,
+        };
+    });
+
```

Comment:
We can move this into a different file `in the future`(not now), to have a more readable and modular code.